server/platform/services/sharedchannel/sync_send_remote.go

```diff
@@ -290,6 +290,10 @@ func (scs *Service) fetchPostsForSync(sd *syncData) error {
 	if err != nil {
 		return fmt.Errorf("could not fetch new posts for sync: %w", err)
 	}
+
+	// Filter out channel metadata system posts that shouldn't be synchronized
+	posts = filterChannelMetadataSystemPosts(posts)
+
```

Comment:
Same, I would move this to the `filterPostsForSync` method that happens right after the `fetchPostsForSync`